bpf/Makefile

```diff
@@ -80,7 +80,7 @@ LB_OPTIONS = \
 # tested as well.
 MAX_BASE_OPTIONS = -DSKIP_DEBUG=1 -DENABLE_IPV4=1 -DENABLE_IPV6=1 \
 	-DENABLE_HOST_SERVICES_TCP=1 -DENABLE_HOST_SERVICES_UDP=1 \
-	-DENABLE_HOST_REDIRECT=1 -DENABLE_ROUTING=1 -DNO_REDIRECT=1 \
+	-DENABLE_HOST_REDIRECT=1 -DENABLE_ENDPOINT_ROUTES=0 -DNO_REDIRECT=1 \
```

Comment:
This is still a definition and the `#ifdef` will pass. I see that you have updated the conditions with `== 0`, but _this_ won't work when the agent does _not_ define the option at all.

You want to make sure the macro is undefined here, check the `-U` option maybe.

Same remark for the other changes in the Makefile.